src/features/labels/Vote.tsx

```diff
@@ -26,45 +28,37 @@ const Container = styled.div<{ vote: 1 | -1 | 0 | undefined }>`
 `;
 
 interface VoteProps {
-  type: "comment" | "post";
-  id: number;
-  score: number;
-  voteFromServer: 1 | -1 | 0 | undefined;
+  item: PostView | CommentView;
   className?: string;
 }
 
-export default function Vote({
-  type,
-  id,
-  voteFromServer,
-  score: existingScore,
-  className,
-}: VoteProps) {
+export default function Vote({ item, className }: VoteProps) {
   const [present] = useIonToast();
   const dispatch = useAppDispatch();
   const votesById = useAppSelector((state) =>
-    type === "comment"
+    "comment" in item
```

Comment:
What does `"comment" in item` mean exactly?

Reply:
It infers that it's a `CommentView`, and Typescript picks this up too.